### DISCUSSION TITLE
Declaring the base User using Beanie

### DISCUSSION THREAD

**Discussion by rozkomnadzor** [fastapi-users/fastapi-users#1217] [Q&A] [answered]
I'm creating a base user model in Beanie as in the documentation (via BeanieBaseUser[PydanticObjectId), but for some reason it doesn't work

If I use BeanieBaseUserDocument an unused empty collection is created in Mongo

Maybe I'm misunderstanding something, could you help me

> Traceback (most recent call last):
>   File "D:\Programs\Python\Python310\lib\multiprocessing\process.py", line 315, in _bootstrap
>     self.run()
>   File "D:\Programs\Python\Python310\lib\multiprocessing\process.py", line 108, in run
>     self._target(*self._args, **self._kwargs)
>   File "D:\Workspace\Python\venv\**\lib\site-packages\uvicorn\_subprocess.py", line 76, in subprocess_started
>     target(sockets=sockets)
>   File "D:\Workspace\Python\venv\**\lib\site-packages\uvicorn\server.py", line 61, in run
>     return asyncio.run(self.serve(sockets=sockets))
>   File "D:\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
>     return loop.run_until_complete(main)
>   File "D:\Programs\Python\Python310\lib\asyncio\base_events.py", line 641, in run_until_complete
>     return future.result()
>   File "D:\Workspace\Python\venv\**\lib\site-packages\uvicorn\server.py", line 68, in serve
>     config.load()
>   File "D:\Workspace\Python\venv\**\lib\site-packages\uvicorn\config.py", line 473, in load
>     self.loaded_app = import_from_string(self.app)
>   File "D:\Workspace\Python\venv\**\lib\site-packages\uvicorn\importer.py", line 21, in import_from_string
>     module = importlib.import_module(module_str)
>   File "D:\Programs\Python\Python310\lib\importlib\__init__.py", line 126, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
>   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
>   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
>   File "D:\Workspace\Python\Work\**\api\main.py", line 12, in <module>
>     from dependencies.auth import auth_backend
>   File "D:\Workspace\Python\Work\**\api\dependencies\auth.py", line 7, in <module>
>     from models.user import User, get_user_manager
>   File "D:\Workspace\Python\Work\**\api\models\user.py", line 15, in <module>
>     class User(BeanieBaseUser[PydanticObjectId]):
> TypeError: 'ModelMetaclass' object is not subscriptable

FastAPI: 0.95.1
FastAPI Users: 11.0.0

**Comment by frankie567** [comment, marked as answer]
Since V11, `User` need to inherit from Beanie's `Document` explicitly. Besides, it's not a generic anymore. So it looks like this:

```py
class User(BeanieBaseUser, Document):
    pass
```

Ref: https://fastapi-users.github.io/fastapi-users/11.0/configuration/databases/beanie/

**Reply by rozkomnadzor** [reply]
Thank you very much, apparently for some reason I was directed to past versions
